core/src/main/scala/com/evolutiongaming/kafka/flow/PartitionFlowConfig.scala

```diff
@@ -33,10 +33,11 @@ import scala.concurrent.duration._
   *   Try committing everything when partition is revoked.
   */
 case class PartitionFlowConfig(
-  triggerTimersInterval: FiniteDuration = 1.second,
-  commitOffsetsInterval: FiniteDuration = 1.minute,
-  recoveryMode: RecoveryMode            = RecoveryMode.ParallelUnbounded,
-  commitOnRevoke: Boolean               = false
+  triggerTimersInterval: FiniteDuration    = 1.second,
+  commitOffsetsInterval: FiniteDuration    = 1.minute,
+  recoveryMode: RecoveryMode               = RecoveryMode.ParallelUnbounded,
```

Comment:
`RecoveryMode.ParallelUnbounded` & `TimersExecutionMode.Unbounded` are those the same?

Reply:
No, they are responsible for completely different parts of the flow and should be configured separately

Reply:
I was more thinking about reusing `ParallelizationMode` in different configs… `ParallelizationMode.Unbounded | Bounded | Sequential` etc…

Reply:
Yes, that would be convenient! Though, they're not the same as the new one doesn't have `Sequential`, and introducing it (and implementing the logic) just for reusability seems weird as sequential mode doesn't make much sense there 🤔 Plus, we lose the scaladoc that's very detailed and descriptive as it's tailored for that specific ADT...  
I need more input on that @edubrovski @Phill101 please

Reply:
`Sequential` = `Bounded(1)`, right?

Reply:
Semantically -- yes. But differs in implementation as it doesn't spawn any fibers at all, even waiting ones (`parTraverse` vs `traverse`)

Reply:
I'd prefer to have separate ADTs.
1. I want kafka-flow to fail fast (compilation error / config parsing error) if a user tries to use `Sequential` for timers execution.
2. It's convenient for users to look at the source code of `PartitionFlowConfig` and see what variants are available.

Reply:
>  But differs in implementation as it doesn't spawn any fibers at all

but does it mean that it **SHOULD** be different in implementation :)

Reply:
Another thing we can do is to have a hierarchy:
```
case object Sequential extends ParallelizationMode
sealed trait Parallel extends ParallelizationMode

object Parallel {
  case object Unbounded extends Parallel
  case class Bounded(n) extend Parallel
}
```

And then use `ParallelizationMode.Parallel` for configuring timers. This way we'll avoid duplication and make invalid value (`Sequential`) unusable.

Reply:
> but does it mean that it SHOULD be different in implementation? :)

I'd prefer not to spawn any fibers whenever it's not absolutely necessary. It looks weird to me having a "sequential" mode configured, but spawning 100_000 fibers, even though 99% of them is waiting at any given moment. I don't think such uniformity in configuration data classes is a good tradeoff for such runtime overhead (which is completely unnecessary!)

> Another thing we can do is to have a hierarchy

I like this idea, it covers both cases. If no other maintainers object, I'd go with this one

Reply:
I've changed the hierarchy to a single ADT, please take a look and let me know your opinion